### ISSUE TITLE
using `RUST_BOOSTRAP_CONFIG` to configure x.py can break `boostrap` tests 

### ISSUE THREAD

**Issue by rowan-sl** [rust-lang/rust#120202]
When setting `RUST_BOOTSTRAP_CONFIG` (for x.py) to a path that doesn't end in a file named `config.toml` (or `config.dist.toml`) the `bootstrap` test `profile_user_dist` fails, due to explicitly asserting that the config file has one of these names

(test is `src/bootstrap/src/tests/config.rs:175`) 

This could be indicitive of a larger issue, as [Config::parse_inner](https://doc.rust-lang.org/stable/nightly-rustc/bootstrap/core/config/config/struct.Config.html#method.parse_inner) which is used in many bootstrap tests seems to rely on a concerning amount of external state (this could cause more issues in the future if not resolved, and/or make bootstrap's tests fail when a users config is invalid)

If this behavior is intentional, there is a bug in `rustc_dev_guide`, because the [recommended shell.nix](https://github.com/rust-lang/rustc-dev-guide/blob/b0ee9ec8fa59a6c7620165e061f4747202377a62/src/building/suggested.md#using-nix-shell) sets `RUST_BOOTSTRAP_CONFIG=/nix/store/some/path/rustc-config` which triggers this bug

Alternately, ./x.py could emit an error if RUST_BOOTSTRAP_CONFIG is set to an invalid path, with a suggestion on how to fix it (though this seems like a questionable fix)

@rustbot label +C-bug +T-bootstrap

**Comment by rowan-sl** [comment]
if https://github.com/rust-lang/rustc-dev-guide/pull/1858 is accepted, this would be somewhat lower priority (though not fixed)

**Comment by onur-ozkan** [comment]
> If this behavior is intentional

No, it's not. It's just a logical bug in the test that I will fix quickly.
